packages/block-library/src/term-name/edit.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	useBlockProps,
+	BlockControls,
+	AlignmentControl,
+	InspectorControls,
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import {
+	ToggleControl,
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { useToolsPanelDropdownMenuProps } from '../utils/hooks';
+import { unlock } from '../lock-unlock';
+
+const { HTMLElementControl } = unlock( blockEditorPrivateApis );
+
+export default function TermNameEdit( {
+	attributes,
+	setAttributes,
+	context: { termId, taxonomy },
+} ) {
+	const { textAlign, tagName = 'div', isLink } = attributes;
+
+	const term = useSelect(
+		( select ) => {
+			if ( ! termId || ! taxonomy ) {
+				return null;
+			}
+			return select( coreStore ).getEntityRecord(
+				'taxonomy',
+				taxonomy,
+				termId
+			);
+		},
+		[ termId, taxonomy ]
+	);
+
+	const termName = term?.name || __( 'Term Name' );
+
+	const blockProps = useBlockProps( {
+		className: clsx( {
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+		} ),
+	} );
+
+	const dropdownMenuProps = useToolsPanelDropdownMenuProps();
+
+	const TagName = tagName;
+
+	let termNameDisplay = termName;
+	if ( isLink ) {
+		const termLink = term?.link || '#';
```

Comment:
I think on of the reasons we don't use the actual links in the editors is about screen readers. In most blocks we use the `#pseudo-link` 'pattern'..

Reply:
Ah thanks, changed this to a pseudo link in https://github.com/WordPress/gutenberg/pull/72129/commits/e8fb069e74a9e44794a60760b66097fd0f1f4c4f.